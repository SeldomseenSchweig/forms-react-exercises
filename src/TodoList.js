import React, {useState} from 'react';
import Todo from './Todo.js';
import NewTodoForm from './NewTodoForm';
import { v4 as uuid } from 'uuid'




const TodoList =()=>{
    const INITIAL_STATE = [
        { 
            id:2,
            text:"Wash the car"
        },
        { 
            id:1,
            text: 'Clean the gutters',

        }
    ]
    const [Todos, setTodos] = useState(INITIAL_STATE);
    const addTodo = (newTodo)=>{
        setTodos(Todos =>([...Todos,{...newTodo, id:uuid()} ]));
    }
    const remove = (e) => {
        
        const id = e.target.parentNode.id;
                    
             setTodos((Todos)=>{ 
                const TodosCopy = [...Todos]
                const indexOfObject = TodosCopy.findIndex( object => object.id === id )
                TodosCopy.splice(indexOfObject, 1);
                return [...TodosCopy]
                



             } )
        }
        
    return (

        <div>
            <NewTodoForm addTodo={addTodo} />
            <h1> Todo List</h1>
            <div>
            
            { Todos.map(({id,text}) =><Todo id={id} text={text} key={id} remove={remove}/> )}
            </div>

            
        </div>
    )



}

export default TodoList;