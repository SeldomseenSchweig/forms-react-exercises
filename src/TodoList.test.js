import React from "react";
import { render, fireEvent } from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect';
import TodoList from "./TodoList";


test("renders without crashing", function (){

    render(<TodoList />);
});

test("matches snapshot", function () {
    const {asFragment} = render(<TodoList />)
    expect (asFragment()).toMatchSnapshot();
});

test("should add new item", ()=>{

    const {queryByText, getByLabelText, queryByTestId} = render(<TodoList />)
    const input = getByLabelText('Todo')
    const button = queryByText('Submit')
    
    expect(queryByText('Walk Dogs')).not.toBeInTheDocument();
    fireEvent.change(input,{target:{value: "Walk Dogs"}});
    fireEvent.click(button);
    expect(queryByText('Walk Dogs')).toBeInTheDocument();
    


})
test("should add new item", ()=>{

    const {queryAllByText, queryByText} = render(<TodoList />)
    const remove = queryAllByText('X')
    expect(queryByText('Wash the car')).toBeInTheDocument();
    fireEvent.click(remove[0]);
    expect(queryByText('Wash the car')).not.toBeInTheDocument();
    


})