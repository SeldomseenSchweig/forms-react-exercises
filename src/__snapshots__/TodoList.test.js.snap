// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="todo"
      >
         Todo
      </label>
      <input
        id="todo"
        name="text"
        placeholder="todo"
        type="text"
        value=""
      />
      <button>
        Submit
      </button>
    </form>
    <h1>
       Todo List
    </h1>
    <div>
      <div
        class="Todo"
        data-testid="Wash the car"
        id="1"
        style="border: 3px solid black; margin: 5%; width: 25%;"
      >
        Wash the car
        <button>
          X
        </button>
      </div>
      <div
        class="Todo"
        data-testid="Clean the gutters"
        id="2"
        style="border: 3px solid black; margin: 5%; width: 25%;"
      >
        Clean the gutters
        <button>
          X
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
